qualang_tools/optimal_weights/StateDiscriminator.py

```diff
@@ -109,7 +110,7 @@ def _get_traces(
                     ]
                 )
         else:
-            raise Exception("unknown correction_method")
+            raise Exception("unknown correction_method. Available correction methods are 'gmm', 'robust', and 'none'.")
```

Comment:
I would not write "none", just `gmm` and `robust`. The docstring saying that the default is `robust` is enough.